src/structures.lisp

```diff
@@ -38,13 +38,6 @@
                    (let ((slot (gethash name (structure-slots type))))
                      (convert-to-foreign value (slot-type slot)))))))
 
-(defun convert-into-foreign-memory (value type ptr)
-  (let ((ptype (parse-type type)))
-    (if (typep ptype 'foreign-built-in-type)
-        value
-        (translate-into-foreign-memory value ptype ptr)))
-  ptr)
```

Comment:
So, we have to be careful here because the definition of `convert-into-foreign-memory` has changed. This version looks wrong to me, but it is this way for some reason and code will break when we change it. We can either tackle that issue now or keep this strange version (but move it to `early-types.lisp`) and deal with that later. What do you feel like doing?